pkg/imageprovider/invalid.go

```diff
@@ -0,0 +1,19 @@
+package imageprovider
+
+import "fmt"
+
+type ImageBuildInvalid struct {
+	err error
```

Comment:
Minor: from the current usage it looks like that this custom error is essentially string based, so rather than composing the `error` interface it could be enough to implement directly the interface (and also the creation will be simplified) - unless there will be a need to really wrap an error